lib/api/core/models/security/role.js

```diff
@@ -45,11 +44,6 @@ Role.prototype.isActionAllowed = function roleIsActionAllowed (request, kuzzle)
     promises = [],
     path = [];
 
-  // explicit actions on internalIndex must be denied if no explicit restriction on internalIndex is set:
-  if (request.input.resource.index === kuzzle.internalEngine.index && !this.allowInternalIndex) {
```

Comment:
wasn't this check useful to disallow the internalEngine index to be accessed. I think this check is not done anywhere else.